extensions/amp-sidebar/1.0/component.js

```diff
@@ -190,45 +191,51 @@ export function SidebarToolbar({
   children,
   ...rest
 }) {
-  const ref = useRef();
-  const [matches, setMatches] = useState(false);
-  const [targetEl, setTargetEl] = useState(null);
+  const ref = useRef(null);
+  const [target, setTarget] = useState(null);
 
   useEffect(() => {
-    const window = ref.current?.ownerDocument?.defaultView;
-    if (!window) {
+    const doc = ref.current?.ownerDocument;
+    if (!doc) {
       return;
     }
 
-    const mediaQueryList = window.matchMedia(mediaQueryProp);
-    const updateMatches = () => setMatches(mediaQueryList.matches);
-    mediaQueryList.addEventListener('change', updateMatches);
-    setMatches(mediaQueryList.matches);
-    return () => mediaQueryList.removeEventListener('change', updateMatches);
-  }, [mediaQueryProp]);
+    const selector = `#${escapeCssSelectorIdent(toolbarTarget)}`;
+    const newTarget = doc.querySelector(selector);
+    setTarget(newTarget);
+  }, [toolbarTarget]);
 
   useEffect(() => {
-    const document = ref.current?.ownerDocument;
-    if (!document) {
+    const element = ref.current;
+    const doc = ref.current?.ownerDocument;
+    if (!doc || !target) {
       return;
     }
 
-    const selector = `#${CSS.escape(toolbarTarget)}`;
-    const newTargetEl = document.querySelector(selector);
-    setTargetEl(newTargetEl);
-  }, [toolbarTarget]);
+    const clone = element.cloneNode(true);
+    const style = doc.createElement('style');
+    style./*OK*/ textContent =
+      `#${toolbarTarget}{display: none;}` +
+      `@media ${mediaQueryProp}{#${toolbarTarget}{display: initial;}}`;
+
+    target.appendChild(clone);
+    target.appendChild(style);
+    return () => {
+      if (target) {
+        clone && target.removeChild(clone);
```

Comment:
How can either of these be falsey?

Reply:
I can't think of a specific use case, but figured it would be good style.  I have removed checks for `clone` and `style` being falsey.  Should we also remove the check for `target`?

Reply:
My understanding is as follows: 

If we early return, then we do not pass the cleanup function so do not need to check for falsiness on `clone` and `style`.  We just `return` not `return () => cleanup ....` (so cleanup is not run)

Also checked with Justin regarding check on `target`, and similarly we would not reach the cleanup function if `target` is not `truthy`

Reply:
Ah you're right, thanks for clarifying, feel free to remove!